core/src/main/java/org/sonarsource/sonarlint/core/container/analysis/filesystem/InputPathCache.java

```diff
@@ -19,34 +19,33 @@
  */
 package org.sonarsource.sonarlint.core.container.analysis.filesystem;
 
+import com.google.common.collect.LinkedHashMultimap;
+import com.google.common.collect.SetMultimap;
 import java.nio.file.Path;
 import java.util.LinkedHashMap;
+import java.util.LinkedHashSet;
 import java.util.Map;
+import java.util.Set;
 import java.util.SortedSet;
 import java.util.TreeSet;
-
 import org.sonar.api.batch.fs.InputDir;
 import org.sonar.api.batch.fs.InputFile;
 import org.sonar.api.batch.fs.internal.DefaultFileSystem;
 import org.sonar.api.batch.fs.internal.FileExtensionPredicate;
-import org.sonar.api.batch.fs.internal.FilenamePredicate;
 import org.sonarsource.api.sonarlint.SonarLintSide;
 
-import com.google.common.collect.LinkedHashMultimap;
-import com.google.common.collect.SetMultimap;
-
 @SonarLintSide
 public class InputPathCache extends DefaultFileSystem.Cache {
 
-  private final Map<Path, InputFile> inputFileCache = new LinkedHashMap<>();
+  private final Set<InputFile> inputFileCache = new LinkedHashSet<>();
   private final Map<Path, InputDir> inputDirCache = new LinkedHashMap<>();
   private final SetMultimap<String, InputFile> filesByNameCache = LinkedHashMultimap.create();
   private final SetMultimap<String, InputFile> filesByExtensionCache = LinkedHashMultimap.create();
   private final SortedSet<String> languages = new TreeSet<>();
 
   @Override
   public Iterable<InputFile> inputFiles() {
-    return inputFileCache.values();
+    return inputFileCache;
   }
 
   public Iterable<InputDir> allDirs() {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this use of "InputDir"; it is deprecated. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)